### DISCUSSION TITLE
Container that changes its size within a given viewport-size.

### DISCUSSION THREAD

**Discussion by michaelhuber1937** [mantinedev/mantine#1560] [Q&A] [answered]
Is there a possiblity to query the current viewport-size in terms of "xs", "sm", "md", "lg", "xl"?

I want to implement elegently a custom-container that changes its size within a specific viewport-size. Something like this:

```
const currentBreakpoint = useBreakpoint(); // returns "xs" for example

<MyContainer size={breakpoint} sizes={{ xs: 100, sm: 200, lg: 400, xl: 500 }}>
	Content
</MyContainer>


const MyContainer = ({children, ...rest}) => {
	return (
		<Container {...rest}>
			{children}
		</Container>
	);
};
```

I already implemented a solution with the MediaQuery-component but this is a very ugly code. I am coming from MUI and I am wondering why there is no container that handles this by default? I think this is a pretty common use-case.

Maybe there is a another solution? I am new to Mantine so maybe I don't know an appropriate way to do that task.

**Comment by rtivital** [comment, marked as answer]
I recommend creating custom component and use https://mantine.dev/theming/responsive/#media-queries-in-createstyles for this, it is much cleaner that MediaQuery.

**Reply by octalpixel** [reply]
Out of curiosity, what was deciding factor to not use array/object way (like other libraries) for responsive styles ?

**Reply by rtivital** [reply]
It requires additional development and maintanance

**Reply by octalpixel** [reply]
Makes sense
